docs/oauth.md

```diff
@@ -0,0 +1,214 @@
+ </pluginOAuthConfig>
+
+```
+# OAuth on Tableau Desktop
+
+Your Desktop sign in Dialog will look like this, you can choose different auth mode and notice different required fields appear. This example has multiple auth type supported and one of them is oauth. 
+
+![Image](../assets/oauth-desktop-dialog.png)
+
+By clicking the sign in button, you will be directed to your OAuth Provider's sign in Url in a default browser on your machine where you can input your credentials, Tableau will get the required attrs back and upon complete you will see this screen:
+
+![Image](../assets/oauth-desktop-complete.png)
+
+# OAuth on Tableau Server
+
+## Configure OAuth Clients on Server
+
+Config your OAuth client on your server: This would be the first step you need to perform for enabling OAuth on Tableau Server, this will setup OAuth Client information to be used on Tableau Server, e.g.:
+```
+tsm configuration set -k oauth.config.clients -v "[{\"oauth.config.id\":\"snowflake_oauth\", \"oauth.config.client_id\":\"[your_client_id]\", \"oauth.config.client_secret\":\"[your_client_secret]\", \"oauth.config.redirect_uri\":\"[your_redirect_url]\"}]" --force-keys
+```
+You need to subsitute [your_client_id], [your_client_secret], [your_redirect_url] with the ones you registered in your provider's OAuth registration page.
+[your_redirect_url] needs to follow certain format, if your server address is https://Myserver/ then [your_reirect_uri] needs to be https://Myserver/auth/add_oauth_token.
+
+## Server Add OAuth token flow
+
+To add an OAuth credential on Tableau Server, you will go to **My Account Settings** page, look for your class in the **Saved Credentals For DataSources** Section.
+After successfully added your credential you will notice an entry appear under your class.
+
+![Image](../assets//oauth-server-addtoken.png)
+
+## Desktop Publish flow
+
+When publishing a pluggable OAuth Workbook/DataSource to Tableau Server, you will see multiple auth options:
+**Prompt** means this resource will published without embedding credential, viewers would need to provide credential to access the resource.
+**Embedding [your_username]** means you will embed the credential with username **[your_username]** to this resource, so all the viewer can use the same credential **[ABC]** to access the resource. Note in order for this to show up, you must already have added a saved OAuth credential according to previous section. You would see multiple entries if you have multiple records of saved credentials and you can pick which one you wanna use for embedding. 
+**Embed Password** is no longer a supported auth mechanism for Pluggable OAuth connectors and an error will show up if you choose that option.
+
+![Image](../assets/oauth-desktop-publish.PNG)
+
+## Web Create flow
+
+For Web Create, the UI dialog would be same with Tableau Desktop with the difference that we are using the server OAuth Client configs.
```

Comment:
```suggestion
For Web Authoring, the UI dialog will be same as Tableau Desktop. The difference will be that we are using the server OAuth Client configs.
```
I don't follow this part?  There is only one oauthConfig.xml for a connector.  Are different parts of it used on server and not desktop?

Reply:
Nope, since we are not registering the server oauth clients info in the OAuthConfig so on server we'll use the one registered with tsm command, while on desktop we use the one registered in oauthConfig file.

Reply:
Got it, I missed that.  Maybe something like we'll use the server OAuth Client config set above with the tsm command.